app/src/main/java/at/roteskreuz/stopcorona/model/entities/infection/exposure_keys/IndexOfDiagnosisKeysArchives.kt

```diff
@@ -2,24 +2,42 @@ package at.roteskreuz.stopcorona.model.entities.infection.exposure_keys
 
 import com.squareup.moshi.Json
 import com.squareup.moshi.JsonClass
+import org.threeten.bp.ZonedDateTime
+import java.time.DateTimeException
 
 /**
  * Describes infection info about user with data gathered from the Exposure SDK.
  */
 @JsonClass(generateAdapter = true)
 data class IndexOfDiagnosisKeysArchives(
-    @field:Json(name = "full_batch")
-    val fullBatch: DiagnosisKeysBatch,
+    @field:Json(name = "full_14_batch")
+    val full14DaysBatch: DiagnosisKeysBatch,
+
+    @field:Json(name = "full_7_batch")
+    val full07DaysBatch: DiagnosisKeysBatch,
 
     @field:Json(name = "daily_batches")
     val dailyBatches: List<DiagnosisKeysBatch>
-)
+){
+    fun batchesForLastHours(hour: Long): List<DiagnosisKeysBatch>{
+        return dailyBatches.filter {
+            (it.interval * 600) > ZonedDateTime.now().minusHours(hour).toInstant().epochSecond
```

Comment:
It would be better to add new method into `DateTimeExtensions` and use `toRollingStartIntervalNumber`.

Reply:
@dusanjencik

Reply:
Done